pennylane/fourier/spectrum.py

```diff
@@ -57,19 +58,34 @@ def stop_at(obj):
                 # parameters - this makes sure no new parameters
                 # were created that depend on the inputs
                 try:
+                    # we need to create a new tape here derived from the class
+                    # of the original tape (i.e., AutogradQuantumTape)
+                    # and queue the ops from the expansion,
+                    # since the get_parameters(trainable_only=True)
+                    # method only distinguishes trainable params
+                    # if the tape is interface-aware
+                    new_tape = tape.__class__()
                     expanded = obj.expand()
+                    for op in expanded.operations:
+                        getattr(new_tape, "_ops").append(op)
+
                 except NotImplementedError:
-                    raise ValueError(f"Cannot expand {obj}. Aborting the expansion of the tape.")
+                    raise ValueError(f"Cannot expand {obj.name}. Aborting the expansion of the tape.")
 
-                new_inputs = expanded.get_parameters(trainable_only=True)
+                # necessary to extract the parameters of the ops
+                # and get indices of the trainable ones
+                new_tape._update_par_info()
+                new_tape._update_trainable_params()
+                new_inputs = new_tape.get_parameters(trainable_only=True)
```

Comment:
once the new tape is created, we still have to update its parameter list, else it is empty...